lib/particle.js

```diff
@@ -0,0 +1,87 @@
+const { Vec3 } = require("vec3")
+
+module.exports = loader
+
+function loader (registry) {
+  class Particle {
```

Comment:
In my earlier comment I meant you can remove the unnecessary `get` and `set` functions. Something like this will take alot less code :

```js
class Particle {
  constructor(id, position, count = 1, longDistanceRender, extraData) {
    Object.assign(this, registry.particlesById[id]) // Add in static data about particle 
    // Add in general instance fields for this particle
    // https://minecraft-data.prismarine.js.org/?v=1.8&d=protocol#toClient_world_particles
    // and https://minecraft-data.prismarine.js.org/?v=bedrock_1.16.220&d=protocol#packet_spawn_particle_effect
    this.id = id
    this.position = position
    this.count = count
    this.longDistanceRender = longDistanceRender
    Object.assign(this, extraData) // Particle-specific data that can vary by version
  }

  // This doesn't need to be here, it can also be handled in packet event handler
  static fromNetwork(packet) {
     return new Particle(packet.particleId, new Vec(particle.x, ...), packet.particles, packet.longDistance, {
       movementSpeed: packet.particleData,
     })
  }
}
```

Pretty much everything on particle should be well defined and not copied 1:1 from the network without understanding what it is. Otherwise in the future when the packet structures fields change or different editions get supported it's likely something will break.